### PR TITLE
super quick PR with basic validation script

### PR DESCRIPTION
**Pull request by steeleb**
Okay, this is no where near finished, but your feedback on the categories would be helpful. Also, this gives you something to work from for directions/readme.

I'm currently working on programmatically adding some of the image info (specifically date and mission), but it's proving pretty hard as the GROD script was set up. We might need to have the user run the script after every Tile-Date has been labelled. That's obviously not ideal at all, so I'm going to try to figure out how to add properties to the labels given a particular Tile-Date. 

More soon!

### CODE REVIEW

**Comment by lindsayplatt** [comment]
So I may have misspoken before ... it turns out that I might not have access to those assets. I just tried running this script but get the following error:

```
ImageCollection (Error)
Image.load: Image asset 'projects/ee-ross-superior/assets/eePlumB_valSets/LS4-7_eePlumB_val_aoi_10_1990-07-24' not found.
```

**Comment by steeleb** [comment]
Okay, edits to the validation script and the validation have been made! 

1) I fixed the cloud mask (again), as masking FOR water apparently also masked out clouds. The only pixel-level mask is now for 'saturated pixels'. 

2) I still can't figure out how to incorporate info from the current Tile-Date into the points you add manually. Because of this, I edited the validation set such that there were no overlapping AOI tiles. _This means that we'll have to assign users mission-date pairs for creating the training set and that anytime a user starts a new mission-date, they will need to start with a fresh script._ 

3) If you think there are better tiles to include, or this is too many tiles (I think this might be 2 too many tiles, just having gone through the labelling process a few times). Please let me know. I'm very open to changing this, just with the previously-mentioned issue with no overlapping tiles. I'm thinking we should drop 'Tile Date 5' (only open water) and maybe 'Tile Date 3' (it's similar to Tile Date 1).

4) I also added two categories: 1) deep offshore sediment and 2) ruddy sediment. Here's my thinking: if we end up not being able to label blooms, wouldn't it bee cool to know the quality of the sediment?! I feel like these are the 3 primary types of visual sediment in this system. If in doubt, we can always collapse. I welcome your feedback.

5) I need to find a better Superior shapefile. I just used the first one I found, but it doesn't have great accuracy and is missing near-shore areas on the southern coast and sucks at islands and jetties. This is causing a lot of 'shorelineContamination' for many of coast-adjacent tiles and I'm just generally not happy with it. 

I feel like we're close to closing #12, but a lot of it depends on your thoughts about the above.

File: eePlumB/1_user_interface_script/eePlumB_validation
Comment (steeleb) [inline eePlumB/1_user_interface_script/eePlumB_validation:1]: Note this was a rename!

Comment (lindsayplatt) [reply eePlumB/1_user_interface_script/eePlumB_validation:1]: Thanks for pointing that out!

**Review by lindsayplatt — COMMENTED** [review]
Alright, I played around with this in GEE. So fun! I did notice that I can't see your point/geometry categories when adding new points (see screenshot below).

image

Responses to your comment: 

> This means that we'll have to assign users mission-date pairs for creating the training set and that anytime a user starts a new mission-date, they will need to start with a fresh script.

Hmmm this is a bummer :( So what does this mean for us? When we "assign users mission-date pairs", what exactly does that require us to do? Edit the script?

> I'm thinking we should drop 'Tile Date 5' (only open water) and maybe 'Tile Date 3' (it's similar to Tile Date 1).

Yeah, this makes sense to me! I just want to make sure that we point out that it is possible to get a scene that is just all open water, not near the shore. When I look at tiles like that, I feel like I'm going crazy trying to distinguish colors and see something. I'll try to remember to put a note in the validation instructions about that.

> I also added two categories: 1) deep offshore sediment and 2) ruddy sediment. 

Oooo this is fun! Especially since there won't be many blooms to label. What is "ruddy" sediment, though?

> I need to find a better Superior shapefile.

I bet Hilary or Kait Reinl would have a better one!

**Review by lindsayplatt — APPROVED** [review]
I am silly and figured out the categories thing as I was writing instructions :) 

image

**Comment by steeleb** [comment]
WAHOO!

> So what does this mean for us? When we "assign users mission-date pairs", what exactly does that require us to do? Edit the script?

I think it just means that the database for labelers is a list of mission-date pairs (i.e., Landsat 5 2008-07-24, that contain all 16 AOIs). The way that the categories are loaded (as imports) means you can't 'edit' those imports once they are in that top area. I tried for a while to export info, but the only way to do that is to use the reducer function, but then you still can't set all the categories back to zero features. It's pretty frustrating. 

Here's a work around I found, though it's not intuitive at all:
1) import the records by hovering and clicking 'convert'
2) ctrl-z to get the `var` categories back in the script
3) comment out each line where the `var` categories are (/* */ won't work, and if you don't comment it out, the script won't recognize the imports)
4) label pixels and run the output to save to drive
5) delete all the imports
6) uncomment the `var` categories
7) repeat the above steps for each new mission-date

It just seems simpler for a user to 'start a new script' for each mission-date pair instead of mess around with all this.

> What is "ruddy" sediment, though?

Great question! I was seeing this mostly near the harbor, and sometimes on the southern coast or in that inlet in the far right corner of our total AOI. Here are a few examples:
image
image
image

This is definitely 'nuance' and doesn't happen on the shoreline very often, but I am deeply curious if there's something that's happening upstream/climatologically.  I am just really interested in what we can do that is 'new' or 'novel' and characterizing sediment plume extent AND some quality would be really cool and it's new and novel. I will search later for an even better southern shore example for our documentation. 

> I bet Hilary or Kait Reinl would have a better one!

Good call! I'll reach out to Hilary first. :)
